### ISSUE TITLE
Start with windows feature not working

### ISSUE THREAD

**Issue by Quake4Life** [suurjaak/InputScope#28]
Win10 x64
22H2 19045.4170

Traceback (most recent call last):
  File "inputscope\main.py", line 418, in OnToggleStartup
  File "inputscope\main.py", line 506, in start
  File "inputscope\main.py", line 534, in create_shortcut
  File "<COMObject <unknown>>", line 2, in save
pywintypes.com_error: (-2147352567, 'Exception occurred.', (0, 'WshShortcut.Save', 'Unable to save shortcut "C:\\Users\\vovan\\Start Menu\\Programs\\Startup\\InputScope.lnk".', None, 0, -2147024893), None)
Traceback (most recent call last):
  File "inputscope\main.py", line 418, in OnToggleStartup
  File "inputscope\main.py", line 506, in start
  File "inputscope\main.py", line 534, in create_shortcut
  File "<COMObject <unknown>>", line 2, in save
pywintypes.com_error: (-2147352567, 'Exception occurred.', (0, 'WshShortcut.Save', 'Unable to save shortcut "C:\\Users\\vovan\\Start Menu\\Programs\\Startup\\InputScope.lnk".', None, 0, -2147024893), None)
Traceback (most recent call last):
  File "inputscope\main.py", line 418, in OnToggleStartup
  File "inputscope\main.py", line 506, in start
  File "inputscope\main.py", line 534, in create_shortcut
  File "<COMObject <unknown>>", line 2, in save
pywintypes.com_error: (-2147352567, 'Exception occurred.', (0, 'WshShortcut.Save', 'Unable to save shortcut "C:\\Users\\vovan\\Start Menu\\Programs\\Startup\\InputScope.lnk".', None, 0, -2147024893), None)

**Comment by suurjaak** [comment]
Does the folder `C:\Users\vovan\Start Menu\Programs\Startup` itself exist?

If yes, is there already a shortcut named `inputscope.lnk` under it?

If no, what happens if you run the program with elevated permissions?

**Comment by Quake4Life** [comment]
Yes, the folder exists.
No, there is no shortcuts.
Yes, I start the program as an admin.

вт, 19 мар. 2024 г., 21:32 Erki Suurjaak ***@***.***>:

**Comment by suurjaak** [comment]
Interesting.. Can you try the following:

- right-click on the InputScope icon in system tray and select "Show Python console"
- in the opened console window, copy-paste the following lines, one by one, pressing Enter after each:
```python
import win32com.client

shell = win32com.client.Dispatch("WScript.Shell")

path = r"C:\Users\vovan\Start Menu\Programs\Startup\InputScope.lnk"

shortcut = shell.CreateShortCut(path)

shortcut.Targetpath = sys.argv[0]

shortcut.save()
```

What is the result?

**Comment by Quake4Life** [comment]
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "<COMObject <unknown>>", line 2, in save
pywintypes.com_error: (-2147352567, 'Exception occurred.', (0,
'WshShortcut.Save', 'Unable to save shortcut "C:\\Users\\vovan\\Start
Menu\\Programs\\Startup\\InputScope.lnk".', None, 0, -2147024893), None)

ср, 20 мар. 2024 г. в 19:43, Erki Suurjaak ***@***.***>:

**Comment by suurjaak** [comment]
Strange. Because error code -2147024893 means "The system cannot find the path specified.".

What happens if you open the console and execute
```python
os.listdir(r"C:\Users\vovan\Start Menu\Programs\Startup")
```
?

**Comment by Quake4Life** [comment]
FileNotFoundError: [WinError 3] The system cannot find the path specified:
'C:\\Users\\vovan\\Start Menu\\Programs\\Startup'

ср, 20 мар. 2024 г. в 20:56, Erki Suurjaak ***@***.***>:

**Comment by suurjaak** [comment]
Well, it looks like the folder `C:\Users\vovan\Start Menu\Programs\Startup` does not exist? What happens if you choose Run from start menu and paste this folder name there and click OK?

**Comment by shenef** [comment]
Checking on my end (on Win 11), i don't have that folder either, for me the startup folder is at `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup` and, looking online, it should be at that same location in Win 10.

**Comment by Quake4Life** [comment]
Yep, there's no such folder.
But as Shenef has mentioned, I surely have
C:\ProgramData\Microsoft\Windows\Start
Menu\Programs\Startup

чт, 21 мар. 2024 г. в 01:07, shenef ***@***.***>:

**Comment by suurjaak** [comment]
Ok, thank you all, now the confusion is cleared up, InputScope uses the wrong path. Will issue a fix soon. 

@Quake4Life just in case, can you try one more thing in the Python console:
```python
import win32com.client

shell = win32com.client.Dispatch("WScript.Shell")

shell.SpecialFolders("Startup")
```
What does it print?

**Comment by Quake4Life** [comment]
'C:\\Users\\vovan\\AppData\\Roaming\\Microsoft\\Windows\\Start
Menu\\Programs\\Startup'

чт, 21 мар. 2024 г. в 12:50, Erki Suurjaak ***@***.***>:

**Comment by suurjaak** [comment]
The newly released version 1.10 should fix this issue.

**Comment by Quake4Life** [comment]
Thanks, I'll check it out.

вт, 26 мар. 2024 г., 00:33 Erki Suurjaak ***@***.***>:
